### DISCUSSION TITLE
How to specify version of the image

### DISCUSSION THREAD

**Discussion by MMarus** [community/community#160672] [Actions]
In azure pipelines, we can specify the ubuntu version e.g.:

pool:
  vmImage: ubuntu-20.04

However, we are getting a different version of the agent in different stages (the first stage uses Image Version: 20210517.1, and the second stage uses Image Version: 20210524.1)
Is there any possibility to specify also the version of the image, so that we avoid having different versions in different stages?
This is a big problem, for example when we try to do terraform plan in 1st stage and in the 2nd terraform apply. This fails because the versions of terraform are also different.

**Comment by miketimofeev** [comment]
@MMarus unfortunately, it is not possible to pin the image version for a pipeline. But we keep working on speeding up the deployment process to reduce the timeframe when images are different. We receive such issues in this repo from time to time, so we know this long deployment drawback.

We have the idea of storing one previous image that can be selected in critical moments when a pipeline is failing and the customer doesn’t have time to fix it right now, but no ETA for now, it is something that we’d like to have in the future.

**Reply by abmptit** [reply]
We get different behavior between these two images, 
Environment: windows-2019
Version: 20211212.1

Environment: windows-2019
Version: 20211219.1
We would like have to specify the version like that :runs-on: windows-2019:20211212.1

**Reply by miketimofeev** [reply]
@abmptit I'm afraid it is not possible at the moment. Please feel free to file an issue if something is broken for you and we will investigate it.

**Comment by gdotdesign** [comment]
I've run into an issue where pinning the runner version would be useful #4241
